From 9e2a7235e38f1e17d4964d6f18d6556ef0186cf1 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Wed, 19 Feb 2020 16:20:33 +0100
Subject: [PATCH] tests: Fix reading of gboolean into pointer

Otherwise the payload will be lost later on when using GPOINTER_TO_INT
to cast it back.

See: #236
---
 tests/test-device-fake.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/test-device-fake.c b/tests/test-device-fake.c
index cd08537..b6a9ce2 100644
--- a/tests/test-device-fake.c
+++ b/tests/test-device-fake.c
@@ -189,6 +189,7 @@ static void
 fpi_device_fake_capture (FpDevice *device)
 {
   FpiDeviceFake *fake_dev = FPI_DEVICE_FAKE (device);
+  gboolean wait_for_finger;
 
   fake_dev->last_called_function = fpi_device_fake_capture;
   g_assert_cmpuint (fpi_device_get_current_action (device), ==, FPI_DEVICE_ACTION_CAPTURE);
@@ -199,7 +200,8 @@ fpi_device_fake_capture (FpDevice *device)
       return;
     }
 
-  fpi_device_get_capture_data (device, (gboolean *) &fake_dev->action_data);
+  fpi_device_get_capture_data (device, &wait_for_finger);
+  fake_dev->action_data = GINT_TO_POINTER (wait_for_finger);
   fpi_device_capture_complete (device, fake_dev->ret_image, fake_dev->ret_error);
 }
 
-- 
2.24.1
